google/cloud/spanner_v1/_pandas_helpers.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2021 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import warnings
+
+SPANNER_TO_PANDAS_DTYPES = {
+    "INT64": "int64",
+    "STRING": "object",
+    "BOOL": "bool",
+    "BYTES": "object",
+    "ARRAY": "object",
+    "DATE": "datetime64[ns]",
+    "FLOAT64": "float64",
+    "NUMERIC": "object",
+    "TIMESTAMP": "datetime64[ns]",
+}
```

Comment:
Let's remove this whole set of dtype conversion logic. It's error prone and the pandas defaults arguably get most things right.

Reply:
If we do want to keep this, then it should only include the "nullsafe" dtypes defined here: https://github.com/pydata/pandas-gbq/blob/22a6064ee616fbdd14ce2c8bf8bfe1ed7d3b6291/pandas_gbq/gbq.py#L644-L684

Reply:
Basically just float and datetimes

Reply:
Done